src/logworker.cpp

```diff
@@ -120,6 +120,20 @@ namespace g3 {
       return std::unique_ptr<LogWorker>(new LogWorker);
    }
 
+   void LogWorker::removeWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
+      auto bg_removesink_call = [this, sink] {
+         for (auto it = _impl._sinks.begin(); it != _impl._sinks.end(); ) {
+            if ((*it).get() == sink.get()) {
+               it = _impl._sinks.erase(it);
+            } else {
+               ++it;
+            }
+         }
+      };
+      auto token_done = g3::spawn_task(bg_removesink_call, _impl._bg.get());
```

Comment:
Just that depending how you do it, strict unit test, or going through the existing g3 API will mean initiating of  adding / removing sinks might be done in a different thread.  

I.e. when you add a sink then the actual "adding" happens in a different thread then in the client thread that called the logworker API for adding a sink. 

Depending on how you implement the tests it may or may not be an issue.